### PR TITLE
Travis flake8

### PR DESCRIPTION
**Pull request by travipross**
Adds `flake8` to the pre-commit hook, plus fixes a bunch of things that violated flake8 rules.

I'm totally fine if you want to reject this one if you feel it'd be annoying to keep up with though. Note there were a few rule exclusions I added to the `.flake8` config file, as they directly contradicted `black`. 

I do, however, like the checks for things like  unused imports and bad programming practice (bare exception catching, checking `== None` instead of `is None`, etc). 

There are a lot of formatting changes I've made here, and it's not immediately obvious for all of them what `flake8` rule they were violating. If you were curious and wanted to familiarize, I'd recommend checking out the `travis-pre-commit` branch that this PR uses as its base, and then running `flake8` manually.

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[flake8]
+ignore = E501, E302, W503
```
Comment (travipross) [inline .flake8:2]: These error codes come from a few places, but they represent a specific rule violation. `flake8` inherits a bunch of [rules from pycodestyle](https://pycodestyle.pycqa.org/en/latest/intro.html#error-codes) (which all begin with E and W), and extends that list with a [bunch of its own rules](https://flake8.pycqa.org/en/latest/user/error-codes.html#error-violation-codes) that begin with F.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -15,7 +15,7 @@ repos:
...
     hooks:
     -   id: isort
-# -   repo: https://github.com/PyCQA/flake8
-#     rev: 4.0.1
-#     hooks:
-#     -   id: flake8
+-   repo: https://github.com/PyCQA/flake8
+    rev: 4.0.1
+    hooks:
+    -   id: flake8
```
Comment (travipross) [inline .pre-commit-config.yaml:21]: By the way, adding new hooks for `pre-commit` is as easy as updating this file with a new repo and hook id. Next time the pre-commit tool is triggered, it'll fetch any missing packages and run them.

**Comment by travipross** [comment]
Note the codecov report shows a super tiny reduction in test coverage, but that's just a technicality. The same 2 lines in `cogs/rates.py` are currently not covered by tests, but because I deleted a line or two in that file, the percentage has changed.

**Comment by FM-17** [comment]
> There are a lot of formatting changes I've made here, and it's not immediately obvious for all of them what flake8 rule they were violating. If you were curious and wanted to familiarize, I'd recommend checking out the travis-pre-commit branch that this PR uses as its base, and then running flake8 manually.

- checked out and installed `travis-pre-commit`
- installed `requirements.dev.txt` (to get `flake8`)
- ran `make format`
- ran `flake8` manually

I've included a sample of each resulting rule violation below. If this is the expected behavior, feel free to merge.
```
./build/lib/poglink/cogs/rates.py:180:29: F541 f-string is missing placeholders
./build/lib/poglink/cogs/rates.py:174:80: E501 line too long (128 > 79 characters)
./build/lib/poglink/cogs/rates.py:89:88: W605 invalid escape sequence '\.'
./build/lib/poglink/cogs/rates.py:21:1: E302 expected 2 blank lines, found 1
./build/lib/poglink/cogs/rates.py:12:1: F401 'poglink.config.MIN_POLLING_DELAY' imported but unused
./build/lib/poglink/models/bans.py:115:76: W291 trailing whitespace
./build/lib/poglink/bot.py:86:13: F841 local variable 'e' is assigned to but never used
./build/lib/poglink/utils.py:42:1: W293 blank line contains whitespace
./tests/unit/test_cli.py:51:23: E712 comparison to True should be 'if cond is True:' or 'if cond:'
./tests/integration/test_cogs_rates.py:32:9: E722 do not use bare 'except' 
./tests/unit/test_models_rates.py:153:5: F811 redefinition of unused 'rates' from line 5
```

**Comment by travipross** [comment]
> > There are a lot of formatting changes I've made here, and it's not immediately obvious for all of them what flake8 rule they were violating. If you were curious and wanted to familiarize, I'd recommend checking out the travis-pre-commit branch that this PR uses as its base, and then running flake8 manually.
> 
> * checked out and installed `travis-pre-commit`
> * installed `requirements.dev.txt` (to get `flake8`)
> * ran `make format`
> * ran `flake8` manually
> 
> I've included a sample of each resulting rule violation below. If this is the expected behavior, feel free to merge.

Odd, I thought I replied to this first thing this morning but maybe I didn't? Anyway, yes, that looks correct. Those are the style rules that I had either fixed or chosen to ignore in the `.flake8` config file. I didn't ignore much, but mostly just things that `black` had conflicting opinions on (like line length, and a couple line wrapping things).

**Pull request merged**